src/Neo/Ledger/Blockchain.cs

```diff
@@ -292,8 +292,10 @@ private VerifyResult OnNewBlock(Block block)
                 }
 
                 int blocksPersisted = 0;
-                uint extraRelayingBlocks = system.Settings.MillisecondsPerBlock < ProtocolSettings.Default.MillisecondsPerBlock
-                    ? (ProtocolSettings.Default.MillisecondsPerBlock - system.Settings.MillisecondsPerBlock) / 1000
+                // Check if this system's block time is faster than the default
+                bool hasFasterBlockTime = system.GetBlockGenTime().TotalMilliseconds < ProtocolSettings.Default.MillisecondsPerBlock;
+                uint extraRelayingBlocks = hasFasterBlockTime
+                    ? (ProtocolSettings.Default.MillisecondsPerBlock - (uint)system.GetBlockGenTime().TotalMilliseconds) / 1000
```

Comment:
We need to be careful with such changes, it's important. The problem is that `GetBlockGenTime()` now refers to the current block height and the storage. Both height and storage may be upgraded between two subsequent calls to `GetBlockGenTime()`. Hence, I think we need to calulate `GetBlockGenTime()` once, store result in a variable and then reuse this variable. At least within the bounds of a single method, like here.